pkg/balance/balance.go

```diff
@@ -0,0 +1,112 @@
+package balance
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/safex/gosafex/internal/crypto/derivation"
+	"github.com/safex/gosafex/pkg/safex"
+	"github.com/safex/gosafex/pkg/safexdrpc"
+)
+
+// Containing balance status
+type Balance struct {
+	CashUnlocked  uint64
+	CashLocked    uint64
+	TokenUnlocked uint64
+	TokenLocked   uint64
+}
+
+type KeyType = []byte
+
+type Key struct {
+	Public  [32]byte
+	Private [32]byte
+}
+
+type Address struct {
+	SpendKey Key
+	ViewKey  Key
+	Address  string
+}
+
+type Wallet struct {
+	balance Balance
+	Address Address
+	client  *safexdrpc.Client
+	outputs map[derivation.Key]*safex.Txout // Save output keys.
+}
+
+// @todo:  Move this to some config, or recalculate based on response time
+const blockInterval = 100
+
+func (w *Wallet) ProcessBlockRange(blocks safex.Blocks) bool {
+	// @todo Here handle block metadata.
+
+	// @todo This must be refactored due new discoveries regarding get_tx_hash
+	// Get transaction hashes
+	var txs []string
+	for _, blck := range blocks.Block {
+		txs = append(txs, blck.Txs...)
+		txs = append(txs, blck.MinerTx)
+	}
+
+	// Get transaction data and process.
+	loadedTxs, err := w.client.GetTransactions(txs)
+	if err != nil {
+		return false
+	}
+
+	for _, tx := range loadedTxs.Tx {
+		w.ProcessTransaction(tx)
+	}
+
+	return true
+}
+
+func (w *Wallet) GetBalance() (b Balance, err error) {
+	w.outputs = make(map[derivation.Key]*safex.Txout)
+	// Connect to node.
+	w.client = safexdrpc.InitClient("127.0.0.1", 38001)
```

Comment:
This works for now, but daemon logic should be removed from blockchain processing